diff --git a/mesas/sas/cdflib90/CMakeLists.txt b/mesas/sas/cdflib90/CMakeLists.txt
new file mode 100644
index 0000000..87ad6c3
--- /dev/null
+++ b/mesas/sas/cdflib90/CMakeLists.txt
@@ -0,0 +1,28 @@
+cmake_minimum_required(VERSION 3.0)
+
+project(cdflib90 Fortran)
+set(CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/mod)
+include_directories(${CMAKE_Fortran_MODULE_DIRECTORY})
+
+
+add_library(biomath_constants_mod SHARED biomath_constants_mod.f90)
+add_library(biomath_sort_mode SHARED biomath_sort_mod.f90)
+add_library(biomath_interface_mod SHARED biomath_interface_mod.f90 biomath_constants_mod.f90 biomath_sort_mod.f90 biomath_strings_mod.f90)
+add_library(biomath_mathlib_mod SHARED biomath_mathlib_mod.f90 biomath_constants_mod.f90)
+add_library(biomath_sort_mod SHARED biomath_sort_mod.f90 biomath_constants_mod.f90)
+add_library(biomath_strings_mod SHARED biomath_strings_mod.f90 biomath_constants_mod.f90)
+add_library(cdf_aux_mod SHARED cdf_aux_mod.f90 biomath_constants_mod.f90 biomath_interface_mod.f90 zero_finder.f90 biomath_strings_mod.f90 biomath_sort_mod.f90)
+add_library(cdf_beta_mod SHARED biomath_constants_mod.f90 biomath_mathlib_mod.f90 cdf_aux_mod.f90 zero_finder.f90)
+add_library(cdf_binomial_mod SHARED cdf_binomial_mod.f90 biomath_constants_mod.f90 cdf_aux_mod.f90 cdf_beta_mod.f90 zero_finder.f90 biomath_mathlib_mod.f90)
+add_library(cdf_chisq_mod SHARED biomath_constants_mod.f90 cdf_aux_mod.f90 cdf_gamma_mod.f90 zero_finder.f90 biomath_mathlib_mod.f90)
+add_library(cdf_f_mod SHARED biomath_constants_mod.f90 cdf_aux_mod.f90 cdf_beta_mod.f90 zero_finder.f90 biomath_mathlib_mod.f90)
+add_library(cdf_gamma_mod SHARED biomath_constants_mod.f90 biomath_mathlib_mod.f90 cdf_aux_mod.f90 zero_finder.f90)
+add_library(cdf_nc_chisq_mod SHARED biomath_constants_mod.f90 biomath_mathlib_mod.f90 cdf_aux_mod.f90 cdf_chisq_mod.f90 zero_finder.f90 cdf_gamma_mod.f90)
+add_library(cdf_nc_f_mod SHARED biomath_constants_mod.f90 biomath_mathlib_mod.f90 cdf_aux_mod.f90 cdf_beta_mod.f90 cdf_f_mod.f90 cdf_gamma_mod.f90 zero_finder.f90)
+add_library(cdf_nc_t_mod SHARED biomath_constants_mod.f90 biomath_mathlib_mod.f90 cdf_aux_mod.f90 cdf_beta_mod.f90 cdf_normal_mod.f90 cdf_t_mod.f90 zero_finder.f90)
+add_library(cdf_neg_binomial_mod SHARED biomath_constants_mod.f90 cdf_aux_mod.f90 cdf_beta_mod.f90 zero_finder.f90 biomath_mathlib_mod.f90)
+add_library(cdf_normal_mod SHARED biomath_constants_mod.f90 biomath_mathlib_mod.f90 cdf_aux_mod.f90 zero_finder.f90)
+add_library(cdf_poisson_mod SHARED biomath_constants_mod.f90 cdf_aux_mod.f90 cdf_gamma_mod.f90 biomath_mathlib_mod.f90 zero_finder.f90)
+add_library(cdf_t_mod SHARED biomath_constants_mod.f90 biomath_mathlib_mod.f90 cdf_aux_mod.f90 cdf_beta_mod.f90 cdf_normal_mod.f90 zero_finder.f90)
+add_library(zero_finder SHARED biomath_constants_mod.f90)
+
