diff --git a/setup.py b/setup.py
index 49376a9..aa4c6dd 100644
--- a/setup.py
+++ b/setup.py
@@ -43,18 +43,24 @@ config = {
                     },
     'scripts': [],
     'name': 'mesas',
-    'ext_modules': [Extension(name='solve', sources=[add_mod_dir, util.convert_path('mesas/sas/solve.f90')],
-                              include_dirs=[numpy.get_include()],
-                              extra_f90_compile_args=["-Ofast", '-fno-stack-arrays'],
-                              libraries=reversed(cdflibnames))],
+    'ext_modules': [
+        Extension(
+            name='solve',
+            sources=[util.convert_path('mesas/sas/solve.f90')],
+            include_dirs=[numpy.get_include(), "mesas/sas/cdflib90/_build/mod"],
+            library_dirs=["mesas/sas/cdflib90/_build/mod", "mesas/sas/cdflib90/_build"],
+            extra_f90_compile_args=["-Ofast", '-fno-stack-arrays'],
+            libraries=[p.stem for p in Path("mesas/sas/cdflib90").glob("*_mod.f90")],
+        )
+    ],
 }
 
-config['libraries'] = []
-for src in cdflibnames:
-    config['libraries'] += [(
-        src,
-        {'sources': [f'mesas/sas/cdflib90/{src}.f90'], 'extra_f90_compile_args': ["-Ofast"]}
-    )]
+# config['libraries'] = []
+# for src in cdflibnames:
+#     config['libraries'] += [(
+#         src,
+#         {'sources': [f'mesas/sas/cdflib90/{src}.f90'], 'extra_f90_compile_args': ["-Ofast"]}
+#     )]
 #extra_f90_compile_args=["-Ofast", '-fno-stack-arrays'],
 #extra_f90_compile_args=["-fast", '-acc', '-Minfo', '-Mvect=levels:10', '-ta=tesla:cc35'],
 #extra_f90_compile_args=["-fast", '-acc', '-Minfo', '-ta=multicore'],
